ImageFilterEditor/ImageFilterEditor/ViewController.swift

```diff
@@ -0,0 +1,368 @@
+        // MARK: - UIImage -> CGImage -> CIImage and back
+
+        // UIImage -> CGImage
+        guard let cgImage = image.cgImage else { return nil }
+        let ciImage = CIImage(cgImage: cgImage)
+
+        // filter image
+        let filter = CIFilter.vignetteEffect()
+
+        // Set values
+        filter.inputImage = ciImage
+        filter.center = CGPoint(x: CGFloat(sliderA.value), y: CGFloat(sliderB.value))
+        filter.intensity = sliderC.value
+        filter.radius = sliderD.value
+        //        filter.center = CGPoint(x: 100, y: 150)
+
+        // CI -> CG -> UI
+        guard let outputCIImage = filter.outputImage else { return nil }
+
+        guard let outputCGImage = context.createCGImage(outputCIImage, from: CGRect(origin: .zero, size: image.size )) else { return nil }
+
+        return UIImage(cgImage: outputCGImage)
+    }
+
+    func bumpDistort(_ image: UIImage) -> UIImage? {
+        // MARK: - UIImage -> CGImage -> CIImage and back
+
+        // UIImage -> CGImage
+        guard let cgImage = image.cgImage else { return nil }
+        let ciImage = CIImage(cgImage: cgImage)
+
+        // filter image
+        let filter = CIFilter(name: "CIBumpDistortion")!
+
+        // Set values
+        filter.setValue(ciImage, forKey: kCIInputImageKey)
+        let centerVector = CIVector(cgPoint: CGPoint(x: CGFloat(sliderA.value), y: CGFloat(sliderB.value)))
+        filter.setValue(centerVector, forKey: kCIInputCenterKey)
+        //        filter.center = CGPoint(x: 100, y: 150)
+        filter.setValue(sliderC.value, forKey: kCIInputRadiusKey)
+        filter.setValue(sliderD.value, forKey: kCIInputScaleKey)
+
+        // CI -> CG -> UI
+        guard let outputCIImage = filter.value(forKey: kCIOutputImageKey) as? CIImage else { return nil }
+//        guard let outputCIImage = filter.outputImage else { return nil }
+
+        guard let outputCGImage = context.createCGImage(outputCIImage, from: CGRect(origin: .zero, size: image.size )) else { return nil }
+
+        return UIImage(cgImage: outputCGImage)
+    }
+
+    func gaussian(_ image: UIImage) -> UIImage? {
+        // MARK: - UIImage -> CGImage -> CIImage and back
+
+        // UIImage -> CGImage
+        guard let cgImage = image.cgImage else { return nil }
+        let ciImage = CIImage(cgImage: cgImage)
+
+        // filter image
+        let filter = CIFilter.gaussianBlur()
+
+        // Set values
+        filter.inputImage = ciImage
+        filter.radius = sliderA.value
+        //        filter.center = CGPoint(x: 100, y: 150)
+
+        // CI -> CG -> UI
+        guard let outputCIImage = filter.outputImage else { return nil }
+
+        guard let outputCGImage = context.createCGImage(outputCIImage, from: CGRect(origin: .zero, size: image.size )) else { return nil }
+
+        return UIImage(cgImage: outputCGImage)
+    }
+
+    func motion(_ image: UIImage) -> UIImage? {
+        // MARK: - UIImage -> CGImage -> CIImage and back
```

Comment:
My preference is personally to leave `MARK`s for sort of "headings" within a file; IMO maybe this should be a regular comment or maybe a documenting comment (`///`).